client/driver/docker.go

```diff
@@ -2130,13 +2130,15 @@ func authFromHelper(helperName string) authBackend {
 		helper := dockerAuthHelperPrefix + helperName
 		cmd := exec.Command(helper, "get")
 
-		// Ensure that the HTTPs prefix exists
-		if !strings.HasPrefix(repo, "https://") {
-			repo = fmt.Sprintf("https://%s", repo)
+		repoInfo, err := parseRepositoryInfo(repo)
+		if err != nil {
+			return nil, err
 		}
 
-		cmd.Stdin = strings.NewReader(repo)
+		// Ensure that the HTTPs prefix exists
+		repoAddr := fmt.Sprintf("https://%s", repoInfo.Hostname())
```

Comment:
is it safe to assume everyone uses https for internal hubs

Reply:
This is whats passed to the helper. The docs only ever show examples with https, but even some of the major cloud registry helpers like AWS ECR and Google GCR strip the https if it exists comming in and do the look up based solely on the domain name. 

https://github.com/awslabs/amazon-ecr-credential-helper/blob/master/ecr-login/api/client.go#L41-L43

https://github.com/GoogleCloudPlatform/docker-credential-gcr/blob/master/credhelper/helper.go#L252:16